DOCS-2541: Move component APIs (#3504)
viamrobotics/docs: docs/components/motor/_index.md

# SME: Rand
---

Motor components provide an API for operating a motor or getting its current status.

> If you have a physical motor that convert electricity into rotary motion, configure it as a _motor_ component.

Reviewer: [nit]
```suggestion
If you have a physical motor that converts electricity into rotary motion, configure it as a _motor_ component.
```
Also though, what other kind of motor would I have? :D I think we should not remove quite *all* the info from this page.

```suggestion
If you have a device that converts electricity into rotary motion, such as a brushed DC motor, a brushless DC motor, or a stepper motor, configure it as a _motor_ component.

Note that if you have a hobby servo, you should configure it as a [servo component](/components/servo/) instead.
```

Author: ok - can't note something that hasn't been mentioned yet though

```suggestion
If you have a device that converts electricity into rotary motion, such as a brushed DC motor, a brushless DC motor, or a stepper motor, configure it as a _motor_ component.

For a hobby servo, configure it as a [servo component](/components/servo/) instead.
```

## Available models

> To use a motor, you have to add it to your machine's configuration.

Reviewer: ```suggestion
To use a motor, you must add it to your machine's configuration.
```

Go to your machine's **CONFIGURE** page, and add a model that supports your motor.

> The _motor_ component type natively supports brushed DC motors, brushless DC motors, and stepper motors controlled by a variety of [motor drivers](https://www.wellpcb.com/what-is-motor-driver.html).

Reviewer: ```suggestion
Because your machine controls your motor using a [motor driver](https://www.wellpcb.com/what-is-motor-driver.html), you technically need to select a motor model that supports your motor driver rather than the motor itself.
```

> The following list shows you the available arm models.

Reviewer: ```suggestion
The following list shows you the available motor models.
```

## Next steps

For general configuration and development info, see:

{{< cards >}}
{{% card link="/how-tos/configure/" noimage="true" %}}
{{% card link="/how-tos/develop-app/" noimage="true" %}}
{{% card link="/get-started/control-motor/" noimage="true" %}}
{{< /cards >}}

You can also use the motor component with the following services:

> - [motion service](/services/slam/): to move machines or components of machines

Reviewer: ```suggestion
- [motion service](/services/motion/): to move machines or components of machines
```
Really though this seems like a bit of a stretch, since you'd use arm or base with motion but not really motor directly AFAIK

> 
> You can also use the motor component with the following services:
> 
> - [motion service](/services/slam/): to move machines or components of machines
> - [navigation service](/services/navigation/): to navigate with GPS
> - [SLAM service](/services/slam/): for mapping

Reviewer: ```suggestion
```
See comment above; you'd use these services with a base or an arm but not with a motor AFAIK
